packages/server/src/db/utils.js

```diff
@@ -0,0 +1,157 @@
+const newid = require("./newid")
+
+const DocumentTypes = {
+  MODEL: "model",
+  RECORD: "record",
+  USER: "user",
+  AUTOMATION: "automation",
+  LINK: "link",
+  APP: "app",
+  ACCESS_LEVEL: "accesslevel",
+}
+
+exports.DocumentTypes = DocumentTypes
+
+const UNICODE_MAX = "\ufff0"
+
+/**
+ * If creating DB allDocs/query params with only a single input this can be used, this
```

Comment:
What do you mean by a "single input" here?

Reply:
I might rename this, to be honest this is just be breaking out the functionality that I was repeating everywhere, most documents are top level, in that they are just `<TYPE>:<ID>` - rather than repeating the same get params build each time I just created a single function I could curry to build all the utility functions for the different document types.

I'll give it a name that better aligns with what its doing though, something like `buildTopLevelDocumentParams` - there only is one document that doesn't have a top level ID and thats the records, in that to build a record ID you need to model (top level) and the record (lower level).